lldb/source/Commands/CommandObjectDWIMPrint.cpp

```diff
@@ -105,6 +109,50 @@ bool CommandObjectDWIMPrint::DoExecute(StringRef command,
     }
   }
 
+  // For Swift frames, rewrite `po 0x12345600` to use `unsafeBitCast`.
+  //
+  // This works only when the address points to an instance of a class. This
+  // matches the behavior of `po` in Objective-C frames.
+  //
+  // The following conditions are required:
+  //   1. The command is `po` (or equivalently the `-O` flag is used)
+  //   2. The current language is Swift
+  //   3. The expression is entirely a hex value
+  //   4. The hex value passes address/memory sanity checks
+  //
+  // The address sanity checks are:
+  //   1. The hex value is a readable memory address
+  //
+  // Future potential sanity checks:
+  //   1. Accept tagged pointers/values
+  //   2. Verify the isa pointer is a known class
+  //   3. Require addresses to be on the heap
+  std::string modified_expr_storage;
+  // Either Swift was explicitly specified, or the frame is Swift.
+  bool is_swift = false;
+  if (m_expr_options.language == lldb::eLanguageTypeSwift)
+    is_swift = true;
+  else if (m_expr_options.language == lldb::eLanguageTypeUnknown)
+    is_swift = frame && frame->GuessLanguage() == lldb::eLanguageTypeSwift;
+  bool is_po = m_varobj_options.use_objc;
+  if (is_swift && is_po) {
+    if (expr.startswith("0x")) {
+      lldb::addr_t addr;
+      bool is_hex = !expr.drop_front(2).getAsInteger(16, addr);
```

Comment:
It seems odd to peel the 0x off by hand.  getAsInteger(0, addr) will figure out that it's in hex.  I'm not sure why you require this be hex anyway.  Do people often do `po 12345` when they don't mean it to be a pointer address?

Reply:
Good question. When people have asked for this, the request is almost always formed as "can we get `po 0xaddress` to work in Swift. I have no idea how many people use `po 12345`. Maybe we can add that down the line, but as you say `po 0x...` isn't seen as universally a good thing, so maybe we're better off by being limited here.

I agree that the code is odd, but it was intentional to limit non-hex values. Should we limit it to hex addresses, or should we allow decimal addresses too? I'd feel better about allowing decimal if I knew how people were getting valid decimal addresses from, since usually memory addresses are given to users in hex.

Reply:
You've already limited the code to hex with the `startswith` check.  Stripping off the 0x and passing 16 seems like overkill...

We try to show pointers in hex if we know they are pointers, but if people are digging them out of memory or maybe they have logging printing that shows object addresses and those might be or might not be in hex.

Also it seems like we're saying "po of a hex literal doesn't do anything good if the literal is treated as such, so let's treat it as a pointer to a class".  But it's equally true that "po of any numerical literal doesn't..." so I'm not sure what we gain by restricting ourselves to hex.

Reply:
Also, po in ObjC will work with either hex or decimal numbers, so it seems odd to introduce this asymmetry.

Reply:
If I had reason to believe people used `po <decimal>` then I would agree. But since I don't know if that form is used, I am inclined to not add it right away and see if anyone asks for it. Again, I don't know of any tools/commands that produce decimal addresses which people would feed back into `po`. Which is the worse sin here, having asymmetry with objc, or providing functionality that we don't know is used (and don't feel great about)?

Reply:
I would instead be inclined to ask instead: "do we have a reason to prevent the use of decimal values for addresses".  We're going to do something not useful with hex ones that don't happen to be pointer values, and then we're going to do something not useful with ALL decimal numbers.  So also doing something useful with the decimal ones that happen to be pointers seems like a better option.  And as a user, would you expect that for `po` for swift, hex numbers mean pointers but the equivalent decimal value doesn't?  That's nowhere in the language, it's just some rule we made up.  And either we're going to tell people about this made up rule by giving an error:

(lldb) po 123454
error: pointer value not in hex, please hex encode it and try again

but that just seems like trolling, or we're going to just let people guess why this didn't do what it really seems like it should do.

Reply:
Remember, people get pointer values from many places, including a printf somebody may have put in their code that didn't use hex for reasons we don't get to make judgements about.

Reply:
Ok, now updated to support decimal too. Thanks.